service/frontend/workflowHandler.go

```diff
@@ -4358,6 +4387,7 @@ func checkPermission(
 	return nil
 }
 
+// this is a clear sign that we should accept a func rather than an interface with a single method
```

Comment:
or just a string.

I'm debating changing it everywhere to just be `allow(true, theRequest.GetDomain())` because that's the same but doesn't involve passing a giant object to something that only needs a tiny piece.